classes/image_populate_mfgtool.bbclass

```diff
@@ -0,0 +1,142 @@
+do_populate_mfgtool[nostamp] = "1"
+do_populate_mfgtool[recrdeptask] += "do_deploy"
+do_populate_mfgtool[depends] += "uuu-bin-native:do_populate_sysroot"
+
+python () {
+    # Process MFGTOOLSCRIPT and handle its fields.
+    mfgtoolscript_flags = d.getVarFlags('MFGTOOLSCRIPT') or {}
+    mfgtoolscript_depends = []
+    mfgtoolscript_deploy_files = []
+    if mfgtoolscript_flags:
+        mfgtoolscript = (d.getVar('MFGTOOLSCRIPT') or "").split()
+        pn = d.getVar("PN")
+
+        for flag, flagval in sorted(mfgtoolscript_flags.items()):
+            items = flagval.split(",")
+            num = len(items)
+            if num > 2:
+                bb.error("%s: MFGTOOLSCRIPT[%s] Only \"depends,deploy files\" can be specified!" % (pn, flag))
+
+            if flag in mfgtoolscript:
+                if num >= 2 and items[1]:
+                    mfgtoolscript_deploy_files.append(items[1])
+                if num >= 1 and items[0]:
+                    mfgtoolscript_depends.append(items[0])
+
+        d.appendVarFlag('do_populate_mfgtool', 'depends', ' ' + ' '.join(mfgtoolscript_depends))
+
+    d.setVar('MFGTOOLSCRIPT_DEPLOY_FILES', ' '.join(mfgtoolscript_deploy_files))
+}
+
+python do_populate_mfgtool() {
+    # For MFGTOOLSCRIPT items we use BitBake's fetcher module allowing a consistent behavior.
+    mfgtoolscript = (d.getVar('MFGTOOLSCRIPT') or "").split()
+    src_uri = ["file://%s" % f for f in mfgtoolscript]
+    if not src_uri:
+        bb.fatal("MFGTOOLSCRIPT is empty so populate_mfgtool cannot be run.")
+        return
+    bb.debug(1, "following script are used: %s" % ', '.join(mfgtoolscript))
+
+    localdata = bb.data.createCopy(d)
+    filespath = (d.getVar('MFGTOOL_FILESPATH') or "")
+    localdata.setVar('FILESPATH', filespath)
+
+    try:
+        fetcher = bb.fetch2.Fetch(src_uri, localdata)
+        fetcher.unpack(localdata.getVar('WORKDIR'))
+    except bb.fetch2.BBFetchException as e:
+        bb.fatal("BitBake Fetcher Error: " + repr(e))
+
+    # Generate MFGTOOL bundle.
+    bb.build.exec_func('generate_mfgtool_bundle', d)
+}
+
+generate_mfgtool_bundle() {
+    bbnote "Processing uuu files ..."
+    for src in $(ls -1 ${WORKDIR}/*.uuu.in); do
+        dest=$(echo $src | sed 's,.in$,,g')
+        bbnote " - $src -> $dest"
+        sed -e 's/@@MACHINE@@/${MACHINE}/g' \
+            -e "s,@@UBOOT_BINARY@@,${UBOOT_BINARY},g" \
+            -e "s,@@SPL_BINARY@@,${SPL_BINARY},g" \
+            -e "s,@@IMAGE_BASENAME@@,${IMAGE_BASENAME},g" \
+            $src > $dest
+    done
+
+    bbnote "Deploying uuu files ..."
+    for src in $(ls -1 ${WORKDIR}/*.uuu); do
+        dest=$(basename $src)
+        bbnote " - $src -> ${MFGTOOLDIR}/${PN}-${MACHINE}/$dest"
+        install -D -m 0644 $src ${MFGTOOLDIR}/${PN}-${MACHINE}/$dest
+    done
+
+    bbnote "Copying uuu binaries..."
+    cp -v -s ${STAGING_LIBDIR_NATIVE}/uuu/* ${MFGTOOLDIR}/${PN}-${MACHINE}/
+
+    bbnote "Copyng MFGTOOL extra deploy files..."
```

Comment:
`s/Copyng/Copying/`

With that and previous notes from other colleagues - LGTM.

Reply:
Fixed.